processing/src/main/java/org/apache/druid/segment/UnnestStorageAdapter.java

```diff
@@ -101,7 +101,7 @@ public Sequence<Cursor> makeCursors(
           Cursor retVal = cursor;
           ColumnCapabilities capabilities = cursor.getColumnSelectorFactory().getColumnCapabilities(dimensionToUnnest);
           if (capabilities != null) {
-            if (capabilities.isDictionaryEncoded().and(capabilities.areDictionaryValuesUnique()).isTrue()) {
+            if (!capabilities.isArray() && capabilities.isDictionaryEncoded().and(capabilities.areDictionaryValuesUnique()).isTrue()) {
```

Comment:
so, this array column is dictionary encoded, but is not dictionary encoded in the correct way for the unnest dimension cursor, instead the dictionary ids stored in the column are the entire arrays id. Unnest will try to incorrectly use a dimension selector for these array columns.

Are there things legitimately typed as ARRAY but should use a dimension selector?

Reply:
i confirmed that this doesn't impact `MV_TO_ARRAY` when used in SQL because the planner unwraps it leaving just a direct column access for the underlying multi-value column, so the capabilities are `STRING` typed with `hasMultipleValues` set to true.